builtin/credential/approle/path_role.go

```diff
@@ -2390,6 +2395,10 @@ func (b *backend) handleRoleSecretIDCommon(ctx context.Context, req *logical.Req
 	// Check whether or not ttl is defined, otherwise fallback to secret_id_ttl
 	if ttlRaw, ok := data.GetOk("ttl"); ok {
 		ttl = time.Second * time.Duration(ttlRaw.(int))
+		// If the ttl is less than the role's secret_id_ttl, throw an error rather than implicitly overriding
+		if ttl < role.SecretIDTTL {
+			return logical.ErrorResponse("ttl cannot be shorter than the role's secret_id_ttl"), nil
```

Comment:
Wouldn't this scenario make sense though? If the default TTL on the role was 90 days, and the client requested a secret-id with a TTL of 5 minutes for some process, then the shorter TTL should win. Just like num_uses, the role config would serve as the upper bound?

Reply:
You are 100% correct. I’ve gone ahead and change it.